heron/api/src/java/com/twitter/heron/api/topology/TopologyBuilder.java

```diff
@@ -185,6 +189,50 @@ public BoltDeclarer setBolt(String id, IWindowedBolt bolt, Number parallelismHin
   }
 
   /**
+   * Define a new bolt in this topology. This defines a stateful windowed bolt, intended for stateful
+   * windowing operations. The {@link IStatefulWindowedBolt#execute(TupleWindow)} method is triggered
+   * for each window interval with the list of current events in the window. During initialization of
+   * this bolt (potentially after failure) {@link IStatefulWindowedBolt#initState(State)}
+   * is invoked with its previously saved state.
+   * @param id the id of this component.
+   * This id is referenced by other components that want to consume this bolt's outputs.
+   * @param bolt the stateful windowed bolt
+   * @param <K> Type of key for {@link com.twitter.heron.api.state.HashMapState}
+   * @param <V> Type of value for {@link com.twitter.heron.api.state.HashMapState}
+   * @return use the returned object to declare the inputs to this component
+   * @throws IllegalArgumentException {@code parallelism_hint} is not positive
+   */
+  @SuppressWarnings("rawtypes")
+  public <K extends Serializable, V extends Serializable> BoltDeclarer setBolt(
```

Comment:
I would argue let `IWindowedBolt extends IBolt`, then 
- we no longer need `setBolt(..., IStatefulWindowBolt, ...)` method.
- it follows the convention we used for low-level heron stateful API as: https://github.com/twitter/heron/blob/master/examples/src/java/com/twitter/heron/examples/api/StatefulWordCountTopology.java
- `IStatefulWindowedBolt` should be a type of `IBolt`. This method signature could potentially lead to ambiguous signature matching.

Reply:
IWindowedBolt cannot extend IBolt since the execute API for IBolt is execute(Tuple tuple) but for windowed bolts in needs to be execute(TupleWindow window)